File: objectModel/CSharp/Microsoft.CommonDataModel.ObjectModel/Persistence/Syms/Types/SymsManifestContent .cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

namespace Microsoft.CommonDataModel.ObjectModel.Persistence.Syms.Types
{
    using Microsoft.CommonDataModel.ObjectModel.Persistence.Syms.Models;
    using Newtonsoft.Json;
    using System.Collections.Generic;

    public class SymsManifestContent
    {
        public DatabaseEntity Database { get; set; }
        public List<TableEntity> Entities { get; set; }
        public List<RelationshipEntity> Relationships { get; set; }
    }
    public class SymsDatabasesResponse
    {
        [JsonProperty("items")]
        public List<DatabaseEntity> Databases { get; set; }
    }

    public class SymsRelationshipResponse
    {
        [JsonProperty("items")]
        public List<RelationshipEntity> Relationships { get; set; }
    }
}

File: objectModel/Python/cdm/persistence/syms/types/syms_manifest_content.py

from cdm.utilities import JObject
from typing import Union, List

from cdm.persistence.syms.models import TableEntity, DatabaseEntity, RelationshipEntity
class SymsManifestContent:
    """The SymsManifestContent use refer syms content."""
    def __init__(self, database = None, entities = None, relationships = None, intial_sync = False,
                 removed_entities = None, removed_relationships = None):
        self.database = database  # type: database_entity
        self.entities = entities  # type: List[table_entity]
        self.relationships = relationships  # type: List[relationship_entity]
        self.intial_sync = intial_sync  # type: bool
        self.removed_entities = removed_entities  # type: List[str]
        self.removed_relationships = removed_relationships  # type: List[str]

class SymsDatabasesResponse(JObject):
    _attribute_map = {
        'databases': {'key': 'items', 'type': '[object]'},
    }
    def __init__(self, items = None):
        self.databases = items  # type: List[DatabaseEntity]

class SymsTableResponse(JObject):
    def __init__(self):
        self.entities = None  # type: List[table_entity]

class SymsRelationshipResponse(JObject):
    def __init__(self):
        self.relationships = None  # type: List[relationship_entity]